### DISCUSSION TITLE
GPIO write

### DISCUSSION THREAD

**Discussion by marcgauch** [arendst/Tasmota#18976] [Q&A] [answered]
*tl;dr: How to write the GPIO?*

Hello

I have a [Sonoff Basic R2](https://tasmota.github.io/docs/devices/Sonoff-Basic/#sonoff-basic-r2) and flashed it with Tasmota 13 (but it could be any other version).

I have the following case:
I have a small ice machine that does not start when I plug it in. So, I have to press a button once to start it and then when I don't want it running any more, I have to press the same button again.
Now I can use my flashed Sonoff to power on the device, but it does not start the ice machine. I have already opened it and saw that I could just solder an optocoupler in place of the button. But my question now is: How would I switch that optocoupler? I somehow miss the terminology to do a proper search in the manual, discussions here, google and YouTube. All I can find is instructions on how to read the state of the GPIOs, but I need to write something. Then I could send an MQTT POWER1 ON to the Sonoff to power on the device and after a few seconds, “click” the optocoupler. For shutting down, I would do the same thing in reverse. Is there a guide for this?

Thank you for even reading this.

**Comment by sfromis** [comment, marked as answer]
If you config a suitable pin as a Relay, you can control it the same way as the existing relay of the Sonoff Basic, indeed using Power commands.

**Reply by marcgauch** [reply]
Thanks a lot

**Comment by marcgauch** [comment]
For people eventually finding this and not having to read: Thanks guys I found it out. So here is what I've done after @sfromis input:
1. Set `SerialLog 0` in the Console
2. In Configuration > Configure Template Create a new Template based on `Generic (18)` where I just entered the values from Sonoff Basic (This is what I understood from the [remark](https://tasmota.github.io/docs/devices/Sonoff-Basic/#sonoff-basic-r2))
3. Saved it
4. In Configuration > Configure Module I choose my Module Type
5. For `D4 GPIO2` I chose Relay and 2
6. Saved it

I controlled it with the multimeter and it changes the voltage from IO2 to GND from 0 (OFF) to 3.3 V (ON) on the Toggle 2.

**Reply by sfromis** [reply]
While it is ok with `SerialLog 0` (you are unlikely to want to use serial logging anyway), that does not make a difference to how you can use gpio 2. Beware that if your circuit somehow can pull down gpio 2 when applying power, this can block the ESP8266 from booting. Of course, that's not a behavior I'd expect from an optocoupler.
